core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -140,7 +146,10 @@ public List<ManifestFile> apply(TableMetadata base) {
         manifest -> GenericManifestFile.copyOf(manifest).withSnapshotId(snapshotId()).build());
     Iterables.addAll(newManifests, appendManifestsWithMetadata);
 
-    if (base.currentSnapshot() != null) {
+    if (parentSnapshotId != null) {
+      newManifests.addAll(base.snapshot(parentSnapshotId).allManifests(ops.io()));
+    }
+    else if (base.currentSnapshot() != null) {
```

Comment:
Pretty sure we won't need this else if. we're always passing in the parent snapshot id. and it will either be the current branch commit, the current table state or null for an empty table (but for the empty table case we wouldn't have the existing manifests so we're not even hitting the branching logic)